jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -712,12 +711,21 @@ public int fill(ByteBuffer buffer) throws IOException
                                     return filled = -1;
 
                                 case BUFFER_UNDERFLOW:
+                                    // Can we compact?
+                                    if (_encryptedInput.position() != 0)
+                                    {
+                                        BufferUtil.compact(_encryptedInput);
```

Comment:
Beware: `compact()` is an optional operation that may not do anything. If that happens, aren't you going to enter an infinite loop?

So shouldn't you check the returned boolean of `BufferUtil.compact` and only continue if it was true?

Reply:
ah that's much better.  Avoids double access of position too!